### PR TITLE
Update example - cities gmap

### PR DESCRIPTION
**Pull request by birdsarah**
The maps_cities example demoing cities over a google map was showing all the
cities offset from the map tile. The problem appears to have been in the
us_cities dataset. This commit adds a new world_cities dataset and updates
the example to use it.

### CODE REVIEW

**Comment by birdsarah** [comment]
One nice thing about this PR is it makes the improvements in #2857 clear.

File: bokeh/sampledata/__init__.py [hunk 1]
```diff
@@ -71,6 +71,7 @@ def download(progress=True):
         (s3, 'gapminder_population.csv'),
         (s3, 'gapminder_life_expectancy.csv'),
         (s3, 'gapminder_regions.csv'),
+        (s3, 'world_cities.zip'),
```
Comment (almarklein) [inline bokeh/sampledata/__init__.py]: Maybe we can remove US_cities.json if  we dont use it and has erroneous data?

**Comment by almarklein** [comment]
Nice, LGTM. How many points are this?

**Comment by almarklein** [comment]
Wait, I am slightly confused. If I understand correctly, overlaid points were always a bit off (whether using webgl or not), and this fixes it (also for webgl), but the us_cities data was *still* off?

It always seemed to me that as you zoomed in, the error would become smaller, which is not what you'd expect if you have a systematic error. Or is this because I was really seeing two errors (points not being in mercator coords + offset in data), and the latter is much smaller than the first?

File: bokeh/sampledata/__init__.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/sampledata/__init__.py]: Agreed but let's do in another PR.

**Comment by birdsarah** [comment]
@almarklein there are two separate things.

1) the web mercator projection, in my simple understanding, gets better and better the more you zoom in. 
2) the us_cities data just seemed to be off by a consistent offset - not sure why.

**Comment by birdsarah** [comment]
I have just had another look at the us_cities data and I'm not seeing an offset (i'm using the new gmap code). I'm not sure what's going on with me, I seem to keep finding problems where there aren't any. I am on a completely new dev environment these last two weeks, I'm starting to think that maybe I'm having some kind of cacheing issue that I haven't had previously that causing me to see problems when there aren't any. I'm sorry for the fuss.  So, we don't need to remove the us_cities data from the repo, because I currently don't think it's inaccurate (maybe someone else wants to check this). @almarklein the world_cities data is in the order of 50,000 points.
